### PR TITLE
[Notifier] [BC BREAK] Final classes

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | no, BC BREAK, but experimental 
| Deprecations? | no 
| Tickets       | ---
| License       | MIT
| Doc PR        | ---

### CODE REVIEW

**Review by nicolas-grekas — COMMENTED** [review]
I'm ":+0:":

My general opinion about `final` is that's an authoritarian keyword that removes power from end users without necessarily easing maintenance - thus with no practical benefit - actually with a real drawback.

Final proved to be useful when we need to add more type hints, but here, all methods are fully typed.

**Comment by OskarStark** [comment]
I agree, but who knows whats coming, and the other bridges have a final keyword + the brdiges are experimental now.

Just my 2cents, feel free to close

**Comment by OskarStark** [comment]
BTW it helps to bring back people here if this class does not fit some of their requirements, instead of just extending the class...

**Review by chalasr — APPROVED** [review]
For consistency at least, others are final already (and I'm fine with that)

**Review by derrabus — APPROVED** [review]
👍🏻 for `final` here. Extending the transport factories does not make sense at all.

File: src/Symfony/Component/Notifier/Bridge/LinkedIn/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+5.3.0
+-----
+
+* [BC BREAK] `LinkedInTransportFactory` is now final
```
Comment (jderusse) [inline src/Symfony/Component/Notifier/Bridge/LinkedIn/CHANGELOG.md:8]: Shouldn't it be reported in `UPGRADE-5.3.md`?

File: src/Symfony/Component/Notifier/Bridge/Zulip/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,12 @@
 CHANGELOG
 =========
 
+5.3.0
+-----
+
+* [BC BREAK] `ZulipTransport` is now final
+* [BC BREAK] `ZulipTransportFactory` is now final
```
Comment (jderusse) [inline src/Symfony/Component/Notifier/Bridge/Zulip/CHANGELOG.md:9]: same

**Review by jderusse — COMMENTED** [review]
Could you please rebase?

**Comment by OskarStark** [comment]
Rebased @jderusse not sure why the lower build fails 🧐

Shall I or not add/move it to the upgrade file? I didn't do that either in the Return Type fix PR, Sorry I don't have a PR number yet, I am currently on a phone 📱

**Review by fabpot — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @OskarStark.

**Pull request merged**